src/lib/draw/EditGeometryMode.svelte

```diff
@@ -0,0 +1,157 @@
+<script lang="ts">
+  import type { LineString, Point, Polygon } from "geojson";
+  import { ButtonGroup, DefaultButton, SecondaryButton } from "lib/govuk";
+  import type { FeatureWithProps } from "lib/maplibre";
+  import { interventionName } from "lib/sidebar/scheme_data";
+  import { gjScheme, mode, pointTool, polygonTool, routeTool } from "stores";
+  import { onDestroy, onMount } from "svelte";
+  import type { Feature, FeatureUnion, Schema } from "types";
+  import PointControls from "./point/PointControls.svelte";
+  import PolygonControls from "./polygon/PolygonControls.svelte";
+  import RouteControls from "./route/RouteControls.svelte";
+  import SnapPolygonControls from "./snap_polygon/SnapPolygonControls.svelte";
+
+  export let schema: Schema;
+  export let id: number;
+
+  let name = "";
+  let controls = "";
+
+  // As a feature is being edited, store the latest version
+  let unsavedFeature: FeatureWithProps<Point | LineString | Polygon> | null =
+    null;
+
+  onMount(() => {
+    let maybeFeature: FeatureUnion | null = null;
+    gjScheme.update((gj) => {
+      maybeFeature = gj.features.find((f) => f.id == id)!;
+      // Hide it from the regular drawing while we're editing
+      maybeFeature.properties.hide_while_editing = true;
+      return gj;
+    });
+    let feature = maybeFeature!;
+    name = interventionName(schema, feature);
+
+    if (feature.geometry.type == "LineString") {
+      $routeTool?.editExistingRoute(feature as Feature<LineString>);
+      $routeTool?.addEventListenerSuccess(onSuccess);
+      $routeTool?.addEventListenerUpdated(onUpdate);
+      $routeTool?.addEventListenerFailure(onFailure);
+      controls = "route";
+    } else if (feature.geometry.type == "Polygon") {
+      if (feature.properties.waypoints) {
+        $routeTool?.editExistingArea(feature as Feature<Polygon>);
```

Comment:
Hmm, the temptation to refactor a bit is hampered slightly by the point tool being an exception to not handling updates. Maybe also the tools could have a single method `setEventListeners(onSuccess, onUpdate, onFailure)`, since there's now not actually any use for them having multiple callbacks at a time. I'll table this for the moment since the explicit logic is working and isolated here, but happy to revisit